### ISSUE TITLE
Inpainting Attack

### ISSUE THREAD

**Issue by Anson-He** [xavihart/Diff-Protect#6]
How can I use your code to attack inpainting?

**Comment by xavihart** [comment]
yes it can partially work, but in-painting is always harder to attack than global SDEdit, especially when the remaining area is small; here is a new work targeting inpainting using data augmentation [1].

I think sds-attack is a plug-and-play trick that can also be applied in [1] to make it faster.

[1] DiffGuard: Semantic Mismatch-Guided Out-of-Distribution Detection using Pre-trained Diffusion Models

**Comment by Anson-He** [comment]
> yes it can partially work, but in-painting is always harder to attack than global SDEdit, especially when the remaining area is small; here is a new work targeting inpainting using data augmentation [1].
> 
> I think sds-attack is a plug-and-play trick that can also be applied in [1] to make it faster.
> 
> [1] DiffGuard: Semantic Mismatch-Guided Out-of-Distribution Detection using Pre-trained Diffusion Models

My question actually is: you show the experiment result of inpainting in your paper using sds-, do you optimize the perturbation by grad*(1-mask) like mist and photoguard or directly by full grad while attacking inpainting? By the way, I think you wanted to share the paper "DiffusionGuard" which was submitted to 2025 ICLR, instead of "DiffGuard", anyway, thanks for your help and congratulations on your outstanding achievements.

**Comment by xavihart** [comment]
ohoh sorry for the wrong reference, you are right.

> you show the experiment result of inpainting in your paper using sds-, do you optimize the perturbation by grad*(1-mask) like mist and photoguard or directly by full grad while attacking inpainting

i did not target inpainting, the attack is exactly the same as attacking global SDEdit

**Comment by Anson-He** [comment]
> ohoh sorry for the wrong reference, you are right.
> 
> > you show the experiment result of inpainting in your paper using sds-, do you optimize the perturbation by grad*(1-mask) like mist and photoguard or directly by full grad while attacking inpainting
> 
> i did not target inpainting, the attack is exactly the same as attacking global SDEdit

No more questions, thanks for your help.
